src/order/rel_classes.lean

```diff
@@ -74,6 +74,22 @@ protected theorem is_asymm.is_irrefl [is_asymm α r] : is_irrefl α r :=
 protected theorem is_total.is_trichotomous (r) [is_total α r] : is_trichotomous α r :=
 ⟨λ a b, or.left_comm.1 (or.inr $ total_of r a b)⟩
 
+instance is_refl_of_irrefl {α : Type u} (r : α → α → Prop) [is_irrefl α r] :
+  is_refl α (λ x y, ¬ r x y) :=
+⟨@is_irrefl.irrefl α r _⟩
+
+instance is_refl_swap_of_irrefl {α : Type u} (r : α → α → Prop) [is_irrefl α r] :
+  is_refl α (λ x y, ¬ r y x) :=
```

Comment:
To me this would make more sense as
```suggestion
lemma is_refl_swap {α : Type u} (r : α → α → Prop) [is_refl α r] :
  is_refl α (λ x y, r y x) :=
```

Reply:
I in fact want the theorems with the negation, but it's a good idea to add the other ones too.